docs/source/howto/data.rst

```diff
@@ -248,11 +247,247 @@ The node class has various methods to set these attributes, such as :py:`~aiida.
+The data in an AiiDA database is stored as a graph of connected entities, which can be easily *queried* with the ``QueryBuilder`` class.
+
+The ``QueryBuilder`` lets you query your AiiDA database independently of the backend used under the hood.
+Before starting to write a query, it helps to:
+
+*   know what you want to query for.
+    In the language of databases, you need to tell the backend what to *project*.
+    For example, you might be interested in the label of a calculation and the PK's of all its outputs.
+*   know the relationships between entities you are interested in.
+    Nodes of an AiiDA graph (vertices) are connected with links (edges).
+    A node can for example be either the input or output of another node, but also an ancestor or a descendant.
+*   know how you want to filter the results of your query.
+
+Once you are clear about what you want and how you can get it, the QueryBuilder will build an SQL-query for you.
+
+There are two ways of using the QueryBuilder:
+
+#.  In the *appender* method, you construct your query step by step using the ``QueryBuilder.append()`` method.
+#.  In the *queryhelp* approach, you construct a dictionary that defines your query and pass it to the ``QueryBuilder``.
+
+Both APIs provide the same functionality - the appender method may be more suitable for interactive use, e.g., in the ``verdi shell``, whereas the queryhelp method can be useful in scripting.
+In this how-to we will focus on the basics of the appender method.
+For more advanced queries or more details on the queryhelp, see the :ref:`topics section on advanced querying <topics:database:advancedquery>`.
+
+.. _how-to:data:find:select:
+
+Selecting entities
+------------------
+
+Using the ``append()`` method of the ``QueryBuilder``, you can query for the entities you are interested in.
+Suppose you want to query for calculation job nodes in your database:
+
+.. code-block:: python
+
+    from aiida.orm.querybuilder import QueryBuilder
+    qb = QueryBuilder()       # Instantiating instance. One instance -> one query
+    qb.append(CalcJobNode)    # Setting first vertex of path
+
+If you are interested in instances of different classes, you can also pass a tuple, list or set of classes.
+However, they have to be of the same ORM-type (e.g. all have to be subclasses of ``Node``):
+
+.. code-block:: python
+
+    qb = QueryBuilder()       # Instantiating instance. One instance -> one query
+    qb.append([CalcJobNode, WorkChainNode]) # Setting first vertice of path, either WorkChainNode or Job.
+
+.. note::
+
+    Remember that :ref:`processes<topics:processes:concepts:types>` have both a run-time ``Process`` that steers them and a ``Node`` that stores their metadata in the database.
+    The QueryBuilder allows you to pass either the ``Node`` class (e.g. ``CalcJobNode``) or the ``Process`` class (e.g. ``CalcJob``), which will automatically apply the correct filters for the type of calculation.
```

Comment:
I'm honestly not quite sure what this exactly means. Can we use either or? Will this lead to the same/different results? Should I prefer one over the other?

Reply:
If I understand correctly, you can indeed use either and it will produce the same result. I suppose ``CalcJobNode`` is better to use, as it is the actual entity you are querying for. If the user *does* use ``CalcJob``, it should not affect their result. Maybe @sphuber can confirm this?

Perhaps this is a detail that is best moved to the topics section. If we do leave it here, maybe the following note is more clear:

```
.. note::

    Remember that :ref:`processes<topics:processes:concepts:types>` have both a run-time ``Process`` that executes them and a ``Node`` that stores their data in the database.
    The QueryBuilder allows you to pass either the ``Node`` class (e.g. ``CalcJobNode``) or the ``Process`` class (e.g. ``CalcJob``), which will automatically select the right entity for the query.
    So, using either ``CalcJobNode`` or ``CalcJob`` will produce the same query results.
```

Reply:
By definition the QueryBuilder can only be used to query for Nodes, or subclasses thereof (and of course the other entities Computer, User, etc.). However, `Process` or any of its subclasses, is not a subclass of `Node`. Therefore, before you could not query for `PwCalculation` in the QueryBuilder. What you had to do instead was:
```
QueryBuilder().append(CalcJobNode, filters={'attributes.process_type': 'aiida.calculations:quantumespresso.pw'})
```
The `process_type` as you can see is the fully qualified entry point name of `PwCalculation`. Since this is cumbersome, I implemented support for appending process classes. This would interpret `append(PwCalculation)` and expand it automatically into the node class and filters as I described.